prow/tide/tide_test.go

```diff
@@ -379,7 +379,111 @@ func (f *fgc) GetCombinedStatus(org, repo, ref string) (*github.CombinedStatus,
 		nil
 }
 
+func TestExpectedStatus(t *testing.T) {
+	neededLabels := []string{"need-1", "need-2", "need-a-very-super-duper-long-label-name"}
+	forbiddenLabels := []string{"forbidden-1", "forbidden-2"}
+	testcases := []struct {
+		name string
+
+		labels []string
+		inPool bool
+
+		state string
+		desc  string
+	}{
```

Comment:
add a testcase for truncation?

Reply:
The second test case checks that.

Reply:
this was less obvious when the truncate comment mentions "keep at least 1", though I guess we don't need a case for keeps more than one...?

Reply:
The test case I had checks that truncation is used when there are multiple labels that would be too long together. I just added another test case to check that at least one label is mentioned, even if the label has a very long name.

Reply:
SGTM, thanks